vllm/engine/arg_utils.py

```diff
@@ -1157,6 +1157,12 @@ def create_engine_config(self,
             if self.enable_chunked_prefill and self.pipeline_parallel_size > 1:
                 raise ValueError("Multi-Step Chunked-Prefill is not supported "
                                  "for pipeline-parallel-size > 1")
+            from vllm.platforms import current_platform
+            if current_platform.is_cpu():
+                logger.warning("Multi-Step (--num-scheduler-steps > 1) is "
+                               "currently not supported for CPUs and has been "
+                               "disabled.")
+                self.num_scheduler_steps = 1
```

Comment:
Technically it's supported because it won't crash. It's fair to have a warning about performance regression, but I don't think we should force disabling multi-step on CPU.

Reply:
> Technically it's supported because it won't crash. It's fair to have a warning about performance regression, but I don't think we should force disabling multi-step on CPU.

Thanks @comaniac for the review.

CPU doesn't support multi-step scheduling.
Please see the code here: https://github.com/vllm-project/vllm/blob/main/vllm/worker/cpu_model_runner.py#L516

Thanks.

Reply:
Hmm, then why you said "Significant performance drop was observed on CPUs with --num-scheduler-steps > 1"?

Reply:
This is because the multi-step logic has been implemented in the scheduler (which is platform independent).

However, the cpu_model_runner (the platform dependent part) doesn't support it yet, which is the root cause of the performance degradation.

Reply:
Although cpu can still run with --num-scheduler-steps > 1, it doesn't run as the design of multi-step scheduling like GPU due to the missing implementation in cpu_model_runner.

Reply:
> I won't call it "performance degradation" if it cannot work out of box (hitting to the exception you pointed out). 
No. It won't hit the exception on CPUs. vLLM will still run but very slow.